### DISCUSSION TITLE
map view now unusable in the android client

### DISCUSSION THREAD

**Discussion by aisbergde** [immich-app/immich#6551] [General]
with the new release (1.93) the map became even more unusable on the Android client. When clicking it to open, about 2 minutes nothing happens, the app is also not responding on anything. Then it appears, but it's build with this strange heat map. And when I try to zoom in, I get a new long delay again.

On the other side: when I open the web client on a smartphone in the mobil browser, the maps works perfectly: it takes a few seconds to get the map, and I have very useful bubbles with numbers inside, instead of the not very useful heat map. Why this kind of map can't be implemented into the mobile app?

**Comment by schuhbacca** [comment]
Just commenting to say I have the opposite experience on Android. It's much faster now in all respects. Not sure what the difference may be.

**Reply by aisbergde** [reply]
How many assets with geo information do you have? Maybe the heat map works fine with a small number of assets. I have nearly 50.000

![Screenshot_2024-01-21-16-45-31-457_com.mi.android.globallauncher.jpg](https://github.com/immich-app/immich/assets/9020187/a954b8bf-37b1-4072-b74b-6ccc906a1021)



I don't like the heat map at all and I don't see any benefit. I don't see how many assets are in a specific area. I can't click to open them all.

![Screenshot_2024-01-21-16-49-00-191_app.alextran.immich.jpg](https://github.com/immich-app/immich/assets/9020187/a3cf7fc2-1919-4048-84b8-c5b7a7c3b291)



Using the web view, is has the nearly perfect user experience using the map for me: fast, I see the number of assets, I can click and get a gallery of the selected bubble. The only thing I still miss is the combination of a map and thumbnails on one page. But I hope, this will be implemented.

**Comment by shenlong-tanwen** [comment]
> with the new release (1.93) the map became even more unusable on the Android client. When clicking it to open, about 2 minutes nothing happens, the app is also not responding on anything. Then it appears, but it's build with this strange heat map. And when I try to zoom in, I get a new long delay again.

Can you share your device specifications and your android version? The map view was updated to fix the performance issues with the previous library and with our testing, the rendering and responsiveness of the new library was much more better than the previous one we were using. For comparison, Can you please apply some map filters such as favorite only, or a particular date range, close the map view and open it again and check how long it takes to load the map now? 

> On the other side: when I open the web client on a smartphone in the mobile browser, the maps works perfectly: it takes a few seconds to get the map, and I have very useful bubbles with numbers inside, instead of the not very useful heat map. Why this kind of map can't be implemented into the mobile app?

I'm sorry that you're not finding the heat-map useful. A vast majority of the people do find it useful so I don't think we'd be removing the heat-map. However, I do have plans to add the clustering from the web into the mobile app as well, so users can choose between whether they want to cluster their assets or to display a heat-map of them.

**Reply by aisbergde** [reply]
The main issue: the map view is not responding, when I call it, for several minutes. I can apply filters only after the call of the maps view. Filtering data resolves the issue (for example the last year will filter < 2000 assets, and the result is displayed fast). But not filtering the data takes to long. Their is no cache. Every time, when I call the map, the app is not responding, until the heat map is built.

My decice
MIUI Global 14.0.5
Android 13

the planned implementation of clustered bubbles will be a good improvement.
